roles/nginxplus/templates/http/library.conf.j2

```diff
@@ -0,0 +1,97 @@
+# Ansible managed
+# local changes will be overwritten with each playbook run
+#
+proxy_cache_path /data/nginx/{{ item.short_name }}/NGINX_cache/ keys_zone={{ item.short_name }}cache:10m;
+
+{% if item.rate_limit is defined %}
+  include /etc/nginx/conf.d/templates/rate-limit-allow-list.conf;
+
+  map $limit $external_traffic {
+      0 "";
+      1 $binary_remote_addr;
+  }
+
+  # zone: 10mb can hold 160K IP addresses in memory
+  limit_req_zone $external_traffic zone={{ item.name }}-ratelimit:{{ item.rate_limit_mb }}m rate={{ item.rate_limit_per_sec }}r/s;
+{% endif %}
+
+upstream {{ item.short_name }} {
+  {% if item.lbtechnique is defined %}
+    {{ item.lbtechnique }};
+  {% endif %}
+    zone {{ item.short_name }} {{ item.zone_size | default("64k") }};
+    {% for name in item.servers %}
+      server {{ name }}.princeton.edu resolve;
+    {% endfor %}
+    sticky learn
+          create=$upstream_cookie_{{ item.short_name }}cookie
+          lookup=$cookie_{{ item.short_name }}cookie
+          zone={{ item.short_name }}client_sessions:1m;
+}
+
+# redirect HTTP traffic to HTTPS
+server {
+    listen 80;
+    server_name {{ item.short_name }}.princeton.edu;
+
+    location / {
+        return 301 https://$server_name$request_uri;
+    }
+}
+
+server {
+    listen 443 ssl;
+    http2 on;
+    server_name {{ item.short_name }}.princeton.edu;
+
+    ssl_certificate            /etc/letsencrypt/live/{{ item.name }}/fullchain.pem;
+    ssl_certificate_key        /etc/letsencrypt/live/{{ item.name }}/privkey.pem;
+    ssl_session_cache          shared:SSL:1m;
+    ssl_prefer_server_ciphers  on;
+    client_max_body_size       0;
+
+  {% if item.visibility == "public" %}
+    location {{ item.location }} {
+      {% if item.app_protect == "enabled" %}
+        app_protect_enable on;
+        app_protect_security_log_enable on;
+      {% endif %}
+        proxy_pass http://{{ item.short_name }};
+        proxy_set_header X-Forwarded-Host $host;
+        proxy_set_header X-Real-IP $remote_addr;
+        proxy_cache {{ item.short_name }}cache;
+        proxy_intercept_errors on;
+        # TODO don't do a health check if there's only one server
+      {% if item.health_check_URI is defined %}
+        health_check interval=10 fails=3 passes=2 uri={{ item.health_check_URI }};
```

Comment:
Could `health_check_URI` have a default of `/` (which I'm guessing is the nginx default) and thus avoid a conditional? I think it's fine either way, just trying to think of ways to simplify.

Reply:
I'm kinda cheating here. We don't want to do health checks on all sites, because for sites with only one server a health check is all overhead and no gain. This way we can use "health_check_URI" tell which sites should have one at all. But yeah, your solution is a good one longer-term.